src/Model/Behavior/SearchableBehavior.php

```diff
@@ -233,9 +233,11 @@ protected function _addCondLike($data, $field) {
 			$this->_config['like'] = ['before' => $this->_config['like'], 'after' => $this->_config['like']];
 		}
 		$field = array_merge($this->_config['like'], $field);
-		if (empty($data[$field['name']])) {
+
+		if (empty($data) || $data[$field['name']] === '') {
```

Comment:
Should be

    if (!isset($data[$field['name']]) || $data[$field['name']] === '') {

I think.